src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/CrossTargetSubscriptionHostBase.cs

```diff
@@ -262,10 +262,12 @@ await subscriber.Value.InitializeSubscriberAsync(this, _activeConfiguredProjectS
 
                     foreach (var subscriber in Subscribers)
                     {
-                        subscriber.Value.AddSubscriptionsAsync(newProjectContext);
+                        Shell.ThreadHelper.JoinableTaskFactory.Run(() => subscriber.Value.AddSubscriptionsAsync(newProjectContext));
```

Comment:
> Is this actually fixing a known race

I am not aware of one,.  Sounds like this is 2 and we should not exit the lock until this subscription is finished.

> Which is why we're using semaphores elsewhere

Can you show me where else in the code we do that? 

> I'd leave it until I rewrite this logic of coordination between configured projects

Fair enough, do we have an issue # for how CPS wants us to do this?